packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -189,9 +201,225 @@ const renderQueryEditorHeaderLabel = (
   return null;
 };
 
+const QueryLoader = observer(
+  (props: {
+    editorStore: QueryEditorStore;
+    applicationStore: ApplicationStore<
+      LegendQueryApplicationConfig,
+      LegendQueryApplicationPlugin
+    >;
+  }) => {
+    const { editorStore, applicationStore } = props;
+    const queryFinderRef = useRef<SelectComponent>(null);
+    const [selectedQueryID, setSelectedQueryID] = useState('');
+    const queryLoaderColumns = ['Name', 'Author', 'Version'];
+    const closeQueryImporter = (): void => {
+      editorStore.setIsQueryLoaderOpen(!editorStore.isQueryLoaderOpen);
+    };
+    const handleEnterQueryImporter = (): void =>
+      queryFinderRef.current?.focus();
+    const loadSelectedQuery = (): void => {
+      if (selectedQueryID) {
+        editorStore.setIsQueryLoaderOpen(!editorStore.isQueryLoaderOpen);
+        applicationStore.navigator.goTo(
+          generateExistingQueryEditorRoute(selectedQueryID),
+        );
+        window.location.reload();
+      } else {
+        applicationStore.notifyWarning(
+          'No query is selected. Please select a query to load or close.',
+        );
+      }
+    };
+    const buildQueryTableRow = (query: LightQuery): string[] => [
+      query.name,
+      query.owner ?? 'anonymous',
+      query.versionId,
+    ];
+    const searchInputRef = useRef<HTMLInputElement>(null);
+    const searchQuery: React.ChangeEventHandler<HTMLInputElement> = (event) => {
+      editorStore.setSearchText(event.target.value);
+      if (editorStore.searchText.length >= 0) {
+        flowResult(editorStore.loadQueries(editorStore.searchText))
+          .then(() => editorStore.setIsLoadingQueryInProgress(false))
```

Comment:
Can we may be abstract this logic in a function because we are using piece of code in three different places?
```
flowResult(editorStore.loadQueries(editorStore.searchText))
          .then(() => editorStore.setIsLoadingQueryInProgress(false))
          .catch(applicationStore.alertUnhandledError);
```